tests/components/recorder/test_init.py

```diff
@@ -135,21 +137,42 @@ def _add_events(hass, events):
         return [ev.to_native() for ev in session.query(Events)]
 
 
+def _state_empty_context(hass, entity_id):
+    # We don't restore context unless we need it by joining the
+    # events table on the event_id for state_changed events
+    state = hass.states.get(entity_id)
+    state.context = Context(id=None)
+    return state
+
+
 # pylint: disable=redefined-outer-name,invalid-name
 def test_saving_state_include_domains(hass_recorder):
     """Test saving and restoring a state."""
     hass = hass_recorder({"include": {"domains": "test2"}})
     states = _add_entities(hass, ["test.recorder", "test2.recorder"])
     assert len(states) == 1
-    assert hass.states.get("test2.recorder") == states[0]
+    assert _state_empty_context(hass, "test2.recorder") == states[0]
+
+
+def test_saving_state_include_domains_globs(hass_recorder):
+    """Test saving and restoring a state."""
+    hass = hass_recorder(
+        {"include": {"domains": "test2", "entity_globs": "*.included_*"}}
+    )
+    states = _add_entities(
+        hass, ["test.recorder", "test2.recorder", "test3.included_entity"]
+    )
+    assert len(states) == 2
+    assert _state_empty_context(hass, "test2.recorder") == states[0]
+    assert _state_empty_context(hass, "test3.included_entity") == states[1]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)